src/nvim/testdir/test_startup.vim

```diff
@@ -227,6 +227,81 @@ func Test_o_arg()
   call delete('Xtestout')
 endfunc
 
+func Test_invalid_args()
+  if !has('unix') || has('gui_running')
+    " can't get output of Vim.
+    return
+  endif
+
+  for opt in ['-Y', '--does-not-exist']
+    let out = split(system(GetVimCommand() .. ' ' .. opt), "\n")
+    call assert_equal(1, v:shell_error)
+    call assert_equal('nvim: Unknown option argument: "' .. opt .. '"', out[0])
+    call assert_equal('More info with "nvim -h"',                       out[1])
+  endfor
+
+  for opt in ['-c', '-i', '-s', '-t', '-u', '-U', '-w', '-W', '--cmd', '--startuptime']
+    let out = split(system(GetVimCommand() .. ' '  .. opt), "\n")
+    call assert_equal(1, v:shell_error)
+    call assert_equal('nvim: Argument missing after: "' .. opt .. '"', out[0])
+    call assert_equal('More info with "nvim -h"',                      out[1])
+  endfor
+
+  if has('clientserver')
+    " FIXME: need to add --servername to this list
+    " but it causes vim-8.1.1282 to crash!
+    for opt in ['--remote', '--remote-send', '--remote-silent', '--remote-expr',
+          \     '--remote-tab', '--remote-tab-wait',
+          \     '--remote-tab-wait-silent', '--remote-tab-silent',
+          \     '--remote-wait', '--remote-wait-silent',
+          \    ]
+      let out = split(system(GetVimCommand() .. ' '  .. opt), "\n")
+      call assert_equal(1, v:shell_error)
+      call assert_match('^VIM - Vi IMproved .* (.*)$',             out[0])
+      call assert_equal('Argument missing after: "' .. opt .. '"', out[1])
+      call assert_equal('More info with: "vim -h"',                out[2])
+    endfor
+  endif
+
+  " FIXME: commented out as this causes vim-8.1.1282 to crash!
+  "if has('clipboard')
+  "  let out = split(system(GetVimCommand() .. ' --display'), "\n")
+  "  call assert_equal(1, v:shell_error)
+  "  call assert_match('^VIM - Vi IMproved .* (.*)$',         out[0])
+  "  call assert_equal('Argument missing after: "--display"', out[1])
+  "  call assert_equal('More info with: "vim -h"',            out[2])
+  "endif
+
+  let out = split(system(GetVimCommand() .. ' -ix'), "\n")
+  call assert_equal(1, v:shell_error)
+  call assert_equal('nvim: Garbage after option argument: "-ix"', out[0])
+  call assert_equal('More info with "nvim -h"',                   out[1])
+
+  " XXX: nvim opens xxx here!?
+  " let out = split(system(GetVimCommand() .. ' - xxx'), "\n")
```

Comment:
yes, this is a Nvim feature. Vim only allows one "editing mode", one of the four listed at `:h vim-arguments`. Nvim allows `-` to be specified while _also_ supporting one of `-t`, `-q`, or `[file ...]`.

So this is not an error in Nvim. See also https://vimways.org/2018/vims-social-life/ 😂

Reply:
Ok, adjusted.

Thanks for the vimways reference - good read!